consensus/polybft/checkpoint_manager.go

```diff
@@ -369,22 +369,24 @@ func getExitEventsFromReceipts(epoch, block uint64, receipts []*types.Receipt) (
 	events := make([]*ExitEvent, 0)
 
 	for i := 0; i < len(receipts); i++ {
-		for _, log := range receipts[i].Logs {
-			if log.Address != contracts.L2StateSenderContract {
-				continue
+		if *receipts[i].Status == types.ReceiptSuccess {
```

Comment:
Because Status is pointer (`Status *ReceiptStatus`), maybe we should check for `nil` value also? 

I am not sure about this and this does not make a difference, but maybe it would be good to revert condition and `continue` if status is not `ReceiptSuccess`? All the code bellow uses `continue` logic